ra/registration-authority.go

```diff
@@ -602,6 +608,12 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(ra.clk.Now()) {
+		err = fmt.Errorf("Expired authorization")
```

Comment:
No, I think 500 is right here.  The WFE should never sent an UpdateAuthorization back to the RA, so if it does, it really is an internal server error.